backend/src/modules/cards/controller/cards.controller.ts

```diff
@@ -134,7 +137,7 @@ export default class CardsController {
 		@Body() deleteCardDto: DeleteCardDto
 	) {
 		const { boardId, cardId } = params;
-		await this.deleteCardApp.delete(boardId, cardId);
+		await this.deleteCardUseCase.execute({ boardId, cardId });
 		this.socketService.sendDeleteCard(deleteCardDto.socketId, deleteCardDto);
```

Comment:
```suggestion
const completionHandler = () => {
		this.socketService.sendDeleteCard(deleteCardDto.socketId, deleteCardDto);
}

```
Instead, you can pass a `completionHandler` to the use-case that executes the function passed as an argument at the end of the task. Here in the controller you can `return this.deleteCardUseCase.execute({ boardId, cardId, completionHandler });` and remove the  `async` method prefix. This is already done on the `boards.controller` namely on the getBoard method